oak_functions/loader/src/server.rs

```diff
@@ -149,25 +148,54 @@ impl FixedSizeBodyPadder for Response {
     }
 }
 
+/// Trait for implementing extensions, to implement new native functionality.
+pub trait OakApiNativeExtension {
+    /// Similar to `invoke_index` in [`wasmi::Externals`], but may return a result to be
+    /// written into the memory of the `WasmState`.
+    fn invoke_index(
+        &self,
+        wasm_state: &WasmState,
+        args: wasmi::RuntimeArgs,
+    ) -> Result<Result<Option<ExtensionResult>, OakStatus>, wasmi::Trap>;
+
+    /// Similar to `resolve_func` in [`wasmi::ModuleImportResolver`], but returns a tuple with a
+    /// function index and a function signature to be converted into [`wasmi::FuncRef`].
+    fn resolve_func(&self) -> anyhow::Result<(usize, wasmi::Signature)>;
+
+    /// Registration info, including a function index and a function name, for registering this
```

Comment:
Is the function index part of the ABI? i.e. if we change the index in the future, does it invalidate previous Wasm modules? My guess is not, since the index is always looked up by name, but I am actually not sure. Could you clarify either way?